kpi-queries/src/main/java/com.c12e.cortex.examples.aggregate/KPIQueries.java

```diff
@@ -0,0 +1,288 @@
+            config = JsonPath.parse(dataSourceConfig);
+            DataSource dataSource = config.read("$", new TypeRef<DataSource>() {});
+            System.out.println(dataSource);
+
+            if (getOrDefault(() -> cortexSession.catalog().getConnection(connection.getProject(), connection.getName()), null) == null) {
+                cortexSession.catalog().createConnection(connection);
+            }
+
+            if (getOrDefault(() -> cortexSession.catalog().getDataSource(dataSource.getProject(), dataSource.getName()), null) == null) {
+                System.out.println("Created the datasource");
+                cortexSession.catalog().createDataSource(dataSource);
+            }
+
+            cortexSession.write()
+                    .dataSource(javaBeanDS.toDF(), project, dataSource.getName())
+                    .mode(SaveMode.Append)
+                    .save();
+        }
+    }
+
+    public void railCommand() {
+        SimpleModule module = new SimpleModule();
+        module.addDeserializer(ProfileSchema.class, new ProfileSchemaDeserializer());
+
+        com.jayway.jsonpath.Configuration.setDefaults(new com.jayway.jsonpath.Configuration.Defaults() {
+            private final JsonProvider jsonProvider = new JacksonJsonProvider(JsonMapper.builder()
+                    .addModules(new KotlinModule.Builder().build(), module)
+                    .build());
+            private final MappingProvider mappingProvider = new JacksonMappingProvider(JsonMapper.builder()
+                    .addModules(new KotlinModule.Builder().build(), module)
+                    .build());
+
+            @Override
+            public JsonProvider jsonProvider() {
+                return jsonProvider;
+            }
+
+            @Override
+            public Set<com.jayway.jsonpath.Option> options() {
+                return EnumSet.noneOf(com.jayway.jsonpath.Option.class);
+            }
+
+            @Override
+            public MappingProvider mappingProvider() {
+                return mappingProvider;
+            }
+
+        });
+    }
+
+    public String buildFilter(String[] cohortFilters, String startDate, String endDate) throws ParseException {
+        StringBuilder filterStr = new StringBuilder("");
+        if (!startDate.isBlank()) {
+            filterStr.append("_timestamp.gte('" + startDate + "')");
+        }
+        if (!endDate.isBlank()) {
+            String filter = "_timestamp.lte('" + endDate + "')";
+            if (filterStr.toString().isBlank()) {
+                filterStr.append(filter);
+            } else {
+                filterStr.append(".and(" + filter + ")");
+            }
+        }
+        if (!String.join("", cohortFilters).isBlank()) {
+            String finalFilter = cohortFilters[0];
+            if (cohortFilters.length > 1) for (int i = 1; i < cohortFilters.length; i = i + 1) {
+                finalFilter = finalFilter + ".or(" + cohortFilters[i] + ")";
+            }
```

Comment:
Interesting way to format the nested `for-loop`. I would lean towards actually nesting it with indentation - minor nitpick